src/meson.build

```diff
@@ -340,6 +340,18 @@ if opt_crypto == 'nolegacy'
   subdir('nolegacy')
   dep_crypto = dependency('', required: false)
 else
+  foreach name : ['eopenssl30', 'eopenssl11', 'openssl11', 'openssl', 'libressl']
+    dep_openssl = dependency(name, version: '>=1.1.0', static: static, required: false)
+    if dep_openssl.found()
+      break
+    endif
+  endforeach
+  if not dep_openssl.found()
+    dep_openssl = dependency('',
+                             required: opt_crypto == 'openssl' or opt_tests.enabled(),
+                             static: static,
+                             fallback: ['openssl', 'openssl_dep'])
+  endif
```

Comment:
This seems like a lot of fairly confusing conditionals. Note that in Meson 0.60.0 you can specify multiple names for the same dependency, and in 0.62.0 (shortly to be released) looking up the name "openssl" specifically, will automatically try a "system" class fallback that probes for the library names and the header with the version.

The former requires bumping your minimum supported version of Meson. The latter is fully backwards compatible as long as you are okay telling people "your weird system will only find openssl if you upgrade Meson, but people with normal systems that have normal pkg-config files will have it work even on older versions of Meson".

Reply:
Maybe put a TODO for now? I wanted to use the multi-name feature, but it's probably best to support whatever is available in most popular Linux distributions (0.56 in Debian stable, 0.55 in RHEL 8, 0.53 in Ubuntu LTS, etc) without forcing users to install meson manually. Package maintainers in particular are not likely to be happy about this.

Let's see what @gsliepen thinks.

Reply:
On the one hand:
- distro maintainers for stable (LTS) distros with older versions of Meson aren't going to package a new version of any software for any reason whatsoever, and thus don't care what version of Meson you use
- regular users on stable (LTS) distros can `pip install` a new version of Meson pretty easily

On the other hand, it might be desirable for users to be able to just have it work using the distro version of Meson.

On the other other hand, depending on which LTS distro is under discussion they may have an end of life version of Python. Meson 0.62.0 drops support for python 3.6.

On the other other other hand, the BSDs are notorious for bundling software in the base system, then refusing to provide pkg-config files for them at all. You have no options for detecting openssl on FreeBSD whatsoever, other than upgrading Meson to 0.62.0 (in fact, FreeBSD is ***exactly*** the reason I added the custom lookup) or falling back to `cc.find_library()` pain. Admittedly that is something you'd have to do with autotools too...

On the other other other other hand, you can just tell people "okay we give up, use the wrapped subproject for OpenSSL". :D That will work for about a dozen versions back.

Reply:
OK, I apologize for wasting your time.

This particular snippet was written way back when `sptps_keypair` and `sptps_speed` required OpenSSL whether you're using it for the two main programs or not.

It was fixed in 77cd819058de43a5fcea54300dde50e03088c318, but I completely forgot about it here.

Having two build systems in parallel is probably not a good idea.

Reply:
Ah, okay, at least then you don't need to fallback to `dependency('', ...)`

:)

Reply:
The new approach for a single use case is pretty clean, and probably the best that can be done without "multiple names" support. At that point it's really just a style difference.